File: Cpp Codes/Remove_CommonChars_Concatenate.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

string RemoveChars(string str1, string str2)
{
    vector<int> hash1(26, 0);
    vector<int> hash2(26, 0);

    int i;
    for(i=0; i<str1.size(); i++)
        hash1[int(str1[i])-97] = 1;
    
    for(i=0; i<str2.size(); i++)
        hash2[int(str2[i])-97] = 1;
    
    string x = "";
    for(i=0; i<str1.size(); i++)
    {
        if(hash2[int(str1[i])-97] != 1)
            x = x+str1[i];
    }

    for(i=0; i<str2.size(); i++)
    {
        if(hash1[int(str2[i])-97] != 1)
            x = x+str2[i];
    }

    if(x.size() == 0)
        return "";
    else
        return x;
}

int main()
{
    int t;
    cout<<"\nEnter the Number of Testcases: ";
    cin>>t;

    while(t)
    {
        string str1, str2;
        cout<<"Enter the Two Strings: ";
        cin>>str1>>str2;

        string ans = RemoveChars(str1, str2);
        if(ans == "")
            cout<<"\nResultant String: "<<-1;
        else
            cout<<"\nResultant String: "<<ans;
        
        t--;
    }
    cout<<"\n";
    return 0;
}

File: Hashing/Remove_CommonChars_Concatenate.py
def RemoveChars(str1, str2):
    hash1 = [0 for k in range(26)]
    hash2 = [0 for k in range(26)]

    for i in range(len(str1)):
        hash1[ord(str1[i])-97] = 1
    
    for i in range(len(str2)):
        hash2[ord(str2[i])-97] = 1
    
    x = ''
    for i in range(len(str1)):
        if hash2[ord(str1[i])-97] != 1:
            x = x+str1[i]
    
    for i in range(len(str2)):
        if hash1[ord(str2[i])-97] != 1:
            x = x+str2[i]
            
    
    if len(x) == 0:
        return -1
    else:
        return x

    
if __name__ == "__main__":
    print("\n")

    # str1 = 'aacdbb'
    # str2 = 'gaffd'
    str1 = 'dkusgfqwniwcbefkpkujoveawvtrmfossimypcxemtgqxnapywaoseq'
    str2 = 'ockiqpwijgu'
    # str1 = 'aabcdpttbc'
    # str2 = 'tbcpabd'

    ans = RemoveChars(str1, str2)
    print(ans)

    print("\n")